test/k8sT/DatapathConfiguration.go

```diff
@@ -222,19 +222,59 @@ var _ = Describe("K8sDatapathConfig", func() {
 	})
 })
 
-func testPodConnectivityAcrossNodes(kubectl *helpers.Kubectl) bool {
-	By("Checking pod connectivity between nodes")
-
-	filter := "zgroup=testDS"
+func fetchPodsWithOffset(kubectl *helpers.Kubectl, name, filter, hostIPAntiAffinity string, callOffset int) (targetPod string, targetPodJSON *helpers.CmdRes) {
```

Comment:
I think this could be replaced by adding anti affinity option to `helpers.Kubectl.ExecInFirstPod` and using that method, or is there anything test specific that I missed.

Reply:
PR #9177 should give a good overview of where I'm trying to get here.

At a high level, I need curl to occur between different nodes. To do this I need to :
* pick a "source" container and determine its host (IP or name), then
* locate a "destination" container (with different labels) that resides on a node that's different from the above, and fetch its IP as well (so I can curl it).

So if I try to use `ExecInFirstPod()`, that could perhaps replace the first bullet point above, but I still need to figure out the command (using the destination IP from the second bullet above) and figure out the target pod / host to ensure that `ExecInFirstPod()` picks a pod on a different node.

Reply:
I think that with some minor changes, the `fetchPodsWithOffset` could probably be moved into the core helpers as a way to "please get me a pod name and json using the specified filter (with optional host antiaffinity option)". More advanced test cases like in the related PR can then pull whatever information they actually need out of the pod json.